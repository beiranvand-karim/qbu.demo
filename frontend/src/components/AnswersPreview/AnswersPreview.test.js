/* global describe, it, jest, expect*/
import {shallow} from "enzyme"
import {AnswersPreview} from "./"
import React from "react"
import renderer from "react-test-renderer"
import {MemoryRouter} from "react-router-dom"

describe('<AnswersPreview />', () => {
   const answerList = [
      {
         "id": 1,
         "userId": 7,
         "selectedOption": 2,
         "questionId": 25,
         "option": "http://localhost:8282/question/25/2.png",
         "questionTitle": "this question title"
      },
      {
         "id": 3,
         "userId": 7,
         "selectedOption": 2,
         "questionId": 25,
         "option": "http://localhost:8282/question/25/2.png",
         "questionTitle": "this question title"
      }
   ];
   const answers = Object.assign({},{data: answerList});

   it('should render without crashing', () => {
      shallow(<AnswersPreview {...answers} />)
   });
   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><AnswersPreview {...answers} /></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});