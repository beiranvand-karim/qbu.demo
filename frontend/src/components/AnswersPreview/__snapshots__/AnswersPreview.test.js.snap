// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswersPreview /> should matches the snapshot 1`] = `
<div>
  <h1>
    here are your answers:
  </h1>
  <div>
    <div
      className="alert alert-primary fade"
      role="alert"
    >
      <h2>
        question title: 
        this question title
      </h2>
      <div>
        <strong>
          you chose: 
        </strong>
        <img
          alt=""
          src="http://localhost:8282/question/25/2.png"
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="alert alert-primary fade"
      role="alert"
    >
      <h2>
        question title: 
        this question title
      </h2>
      <div>
        <strong>
          you chose: 
        </strong>
        <img
          alt=""
          src="http://localhost:8282/question/25/2.png"
        />
      </div>
    </div>
  </div>
</div>
`;
